src/clippoly/include/clippoly.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <clipper2/clipper.h>
+
+#include "glm/ext/vector_float2.hpp"
+#include "public.h"
+
+namespace C2 = Clipper2Lib;
+
+namespace manifold {
+class Clippoly {
+ public:
+  Clippoly();
+  ~Clippoly();
+  Clippoly(const Clippoly& other);
+  Clippoly& operator=(const Clippoly& other);
+  Clippoly(Clippoly&&) noexcept;
+  Clippoly& operator=(Clippoly&&) noexcept;
+  Clippoly(std::vector<glm::vec2> contour);
+  Clippoly(std::vector<std::vector<glm::vec2>> contours);
```

Comment:
Might as well shorthand this to `Clippoly(const Polygons& polygons);`

What do you think of renaming `Clippoly` to `CrossSection`? Then `Polygons` is akin to `Mesh` while `CrossSection` is akin to `Manifold`.

Reply:
Oh right, I forgot about the `idx` that's part of `Polygon`. Maybe that should be removed - it's only really used internally anyway. Does Clipper provide any way to append data to carry along with the edges?

Reply:
Not on edges specifically, but it does support including Z coordinates on points that aren't used, but carried around.

Reply:
CrossSection sounds good. Yea I wasn't sure about whether the indices were getting used or not so I just went along with it 😅.

Reply:
Okay, I've cleaned up our polygons API in #343. Honestly it's a nice simplification anyway, but should also help here. Good to know regarding Z values, but let's ignore them for now.